tools/update-dll-msvc.bat

```diff
@@ -21,29 +18,106 @@ rem \     'unix' : 'gmake',
 rem \    },
 rem \ }
 
+setlocal enabledelayedexpansion
+
 if "%PROCESSOR_ARCHITECTURE%"=="AMD64" (
     set vimproc_arch=64
-    set cpu_arch=AMD64
-    set msvc_arch=amd64
 ) else (
     set vimproc_arch=32
-    set cpu_arch=x86
-    set msvc_arch=x86
 )
 
 set vimproc_dllname=vimproc_win%vimproc_arch%.dll
 
+if exist "%~dp0vswhere.exe" (
+    REM  found the lasted version of Visual C compiler
+    for /f "usebackq tokens=*" %%i in (`"%~dp0vswhere.exe" -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath`) do (
+        set InstallDir=%%i
+
+        if exist "%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt" (
+            set /p msvc_ver=<"%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt"
+
+            rem Trim
+            set msvc_ver=!msvc_ver: =!
+        )
+
+        if exist "%InstallDir%\VC\Auxiliary\Build\vcvars%vimproc_arch%.bat" (
+            call "%InstallDir%\VC\Auxiliary\Build\vcvars%vimproc_arch%.bat"
+        )
+    )
+) else if defined VS140COMNTOOLS (
+    REM  Microsoft Visual Studio 2015 installed
+    set msvc_ver=14
+    call "%VS140COMNTOOLS%\..\..\VC\bin\vcvars%vimproc_arch%.bat"
```

Comment:
So, I suggest:

```suggestion
    call "%VS140COMNTOOLS%\..\..\VC\vcvarsall.bat" %msvc_arch%
```

`%msvc_arch%` should be set to `x86` (for 32-bit) or `x86_amd64` (for 64-bit).

Reply:
@k-takata , the VC 2008 Express edition don't have vcvarsall.bat, I will try to search nearlly all the version of Visual Studio file path for vcvarsall.bat, vcvars32.bat or vcvars64.bat. If you have the list, it is appreciately send to me, so that I can update it.

Reply:
VC 2010 Express:
`%VS100COMNTOOLS%\..\..\VC\vcvarsall.bat`
`%VS100COMNTOOLS%\..\..\VC\bin\vcvars32.bat`

VC 2012 Express:
`%VS110COMNTOOLS%\..\..\VC\vcvarsall.bat`
`%VS110COMNTOOLS%\..\..\VC\bin\vcvars32.bat`
`%VS110COMNTOOLS%\..\..\VC\bin\x86_amd64\vcvarsx86_amd64.bat`

VC 2013 Express:
`%VS120COMNTOOLS%\..\..\VC\vcvarsall.bat`
`%VS120COMNTOOLS%\..\..\VC\bin\vcvars32.bat`
`%VS120COMNTOOLS%\..\..\VC\bin\x86_amd64\vcvarsx86_amd64.bat`

VC 2015 Express:
`%VS140COMNTOOLS%\..\..\VC\vcvarsall.bat`
`%VS140COMNTOOLS%\..\..\VC\bin\vcvars32.bat`
`%VS140COMNTOOLS%\..\..\VC\bin\x86_amd64\vcvarsx86_amd64.bat`

VC 2013/2015 Community also have `VC\bin\amd64\vcvarsamd64.bat` and `VC\bin\amd64\vcvars64.bat`.

Reply:
@k-takata , I have update the method to search the VC installation path, you can review it. I have test it with VC2019 and VC2008, the method is work at my laptop.